figures/views.py

```diff
@@ -322,6 +323,31 @@ def retrieve(self, request, *args, **kwargs):
         return Response(GeneralCourseDataSerializer(course_overview).data)
 
 
+class CourseTopStatsViewSet(CommonAuthMixin, viewsets.ReadOnlyModelViewSet):
+    """
+    Viewset to get top courses by enrollments/completions.
+    """
+    model = CourseDailyMetrics
+
+    # The "kilo paginator"  is a tempoarary hack to return all course to not
+    # have to change the front end until Figures "Level 2"
+    pagination_class = FiguresKiloPagination
+    serializer_class = CourseTopStatsSerializer
+
+    def get_queryset(self):
+        site = django.contrib.sites.shortcuts.get_current_site(self.request)
+        course_ids = figures.sites.get_course_keys_for_site(site)
+        queryset = self.model.objects.filter(course_id__in=course_ids)
+        order_by = self.request.query_params.get('order_by', '')
+        if order_by:
+            order_by_name = order_by.split(',')[0]
+            order_by_sign = order_by.split(',')[1]
+            order_by_sign = '' if order_by_sign == 'asc' else '-'
+            queryset = queryset.order_by(order_by_sign + order_by_name)
+
+        return queryset[:10]
```

Comment:
why are we returning queryset like this?

Reply:
we want to limit top 10 results no?

Reply:
I can make this dynamic as well. Or if we are limiting the results in the frontend, I can remove it.

Reply:
We might need all data for other reports.. I suggest pagination based results to be returned and we can use it on frontend as per our need. Kindly set pagination limit to 10 records per page.